lib/connection.js

```diff
@@ -611,6 +619,9 @@ function channel0(connection) {
     else if (f.id === defs.ConnectionUnblocked) {
       connection.emit('unblocked');
     }
+    else if (f.id === defs.ConnectionUpdateSecretOk) {
+      connection.emit('update-secret-ok');
+    }
```

Comment:
This is where channel 0 accepts the update-secret ok confirmation frame. I decided not to report an error a confirmation is received without the update-secret operation ever having been sent. 

As previously mentioned I used event emitters because I couldn't see how else to invoke the updateSecret callback.

Reply:
This may already have been administered, but the new "secret" can be either a username or JWT to my knowledge.

If JWT, this would pipe into updating whatever authentication mechanism exists that decodes and verifies the new JWT.

I imagine that somewhere in the libs that a new exp (expiration) value is set which is responsible for many of the 403 access denied errors when trying to publish to a connection that has expired.

Reply:
The RabbitMQ docs are pretty light, but the spec only has two arguments...

```json
{
    "id": 70,
    "arguments": [
    {
      "type": "longstr",
      "name": "new-secret"
    },
    {
      "type": "shortstr",
      "name": "reason"
    }
  ],
  "name": "update-secret",
  "synchronous": true
}
```
So I assume it's also handled transparently.